Automattic/jetpack
# External Media: Updates for 8.7

Changes in this PR:
* Handle 401 responses #16041
* Extend button margin to featured image #16059
* Render Media after authentication #16044
* Better reflect accepted media types #16071
* Account for multiple images #16077
* Fix Modal focus loss on arrow keypress #16055
* Add API unit tests #15988
* Improve Insert Flow #16081
* Restore focus on modal close #16102
* Disable for existing galleries #16176
* Improve upload permission check #16185

#### Changes proposed in this Pull Request:

* Adds a handler for 401 responses of the list endpoint.
* Adds margins for "Select Image" button in placeholders and the featured image panel
* Re-renders Google media modal after authentication instead of trying to fill it with a new query
* Updates button test to better reflect accepted media types
* Handles focus updates from arrow key presses
* Adds API unit tests and updates `/copy` permissions check.
* Updated image insert flow.
* Removes external media option from existing galleries.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






It might be easiest to review the merged PRs separately, including their testing instructions.
Generally, these steps should cover most if not all of what was changes here:
* Open the editor and insert the following blocks: image, media/text, gallery, cover
* Notice how the insert button text is appropriate for the types of media they accept
* Notice that there are correct margins (including featured images panel)
* Insert images into the gallery from one of the external libraries.
* Notice the new insert flow that keeps the full modal open and shows it as busy
* If you chose Google Photos, notice how the modal loads recent photos after authentication (not albums)
* After inserting images, notice that external libraries are no longer available to add more images to the gallery


#### Proposed changelog entry for your changes:

* Introducing: External Media!
